assets/js/components/wp-dashboard/wp-dashboard-impressions.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * WPDashboardImpressions component.
+ *
+ * Site Kit by Google, Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from '../../modules/search-console/datastore/constants';
+import { STORE_NAME as CORE_USER } from '../../googlesitekit/datastore/user/constants';
+import { extractSearchConsoleDashboardData, isZeroReport } from '../../modules/search-console/util';
+import { trackEvent } from '../../util/tracking';
+import DataBlock from '../data-block';
+import PreviewBlock from '../PreviewBlock';
+import ReportError from '../ReportError';
+import ReportZero from '../ReportZero';
+const { useSelect } = Data;
+
+const WPDashboardImpressions = () => {
+	const { data, error, loading } = useSelect( ( select ) => {
+		const store = select( STORE_NAME );
+
+		const commonArgs = {
+			dateRange: select( CORE_USER ).getDateRange(),
+		};
+
+		const args = {
+			dimensions: 'date',
+			compareDateRanges: true,
+			...commonArgs,
+		};
+
+		return {
+			data: store.getReport( args ),
+			error: store.getErrorForSelector( 'getReport', [ args ] ),
+			loading: ! store.hasFinishedResolution( 'getReport', [ args ] ),
+		};
+	} );
+
+	if ( ! data || ! data.length ) {
+		return null;
+	}
+
+	const processedData = extractSearchConsoleDashboardData( data );
+
+	const {
+		totalImpressions,
+		totalImpressionsChange,
+	} = processedData;
+
+	if ( loading ) {
+		return <PreviewBlock width="48%" height="92px" />;
+	}
+
+	if ( error ) {
+		trackEvent( 'plugin_setup', 'search_console_error', error.message );
```

Comment:
This should be in a `useEffect` so that it runs only when `error` changes (and when `error` is truth-y).

As it stands this will track an error event on every render (anytime props change). That's not ideal as it could lead to extra events. We'd want something more like:

```js
useEffect(() => {
  if (error) {
    trackEvent( 'plugin_setup', 'search_console_error', error.message );
  }
}, [error]);
```